libs/@blockprotocol/graph/src/shared/stdlib/subgraph/builder.ts

```diff
@@ -151,3 +154,84 @@ export const buildSubgraph = <Temporal extends boolean>(
 
   return subgraph;
 };
+
+/**
+ * Looking to build a subgraph? You probably want {@link buildSubgraph} from `@blockprotocol/graph/stdlib`
+ *
+ * This function will infer/add edges to a subgraph based on the vertices that are already present.
+ * The {@link Subgraph} must not have any edges already present, as this would result in invalid state.
+ * It will add edges for:
+ *  - data type vertices
+ *  - property type vertices
+ *  - entity type vertices
+ *  - entity vertices
+ *
+ * This operation MUTATES the given {@link Subgraph} - you should know why you need to do it.
+ *
+ * @param {Subgraph} subgraph – the subgraph to mutate by adding edges
+ */
+export const inferSubgraphEdges = <Temporal extends boolean>(
+  subgraph: Subgraph<Temporal, EntityRootType<Temporal>>,
+): void => {
+  // Construct object with vertex ids for each vertex kind
+  const vertexIds = typedEntries(subgraph.vertices).reduce(
+    (
+      acc: {
+        dataTypeVertexIds: OntologyTypeVertexId[];
+        propertyTypeVertexIds: OntologyTypeVertexId[];
+        entityTypeVertexIds: OntologyTypeVertexId[];
+        entityVertexIds: EntityVertexId[];
+      },
+      [baseId, entry],
```

Comment:
I find `entry` pretty confusing. Maybe we could call it `mappedRevisions`, `revisionObject`, `revisionVertices`, or something else

Reply:
I agree, `revisionObject` seems fitting here, thanks! fc5a9aa632769336dce86700eac46429da4affe6